cmd/crates/soroban-spec-typescript/src/project_template/src/assembled-tx.ts

```diff
@@ -0,0 +1,706 @@
+    unwrapErr(): E {
+        throw new Error('No error');
+    }
+    unwrap(): T {
+        return this.value;
+    }
+
+    isOk(): boolean {
+        return true;
+    }
+
+    isErr(): boolean {
+        return !this.isOk()
+    }
+}
+
+export class Err<E extends Error_ = Error_> implements Result<any, E> {
+    constructor(readonly error: E) { }
+    unwrapErr(): E {
+        return this.error;
+    }
+    unwrap(): never {
+        throw new Error(this.error.message);
+    }
+
+    isOk(): boolean {
+        return false;
+    }
+
+    isErr(): boolean {
+        return !this.isOk()
+    }
+}
+
+export const contractErrorPattern = /Error\(Contract, #(\d+)\)/;
+
+type AssembledTransactionOptions<T = string> = MethodOptions &
+  ClassOptions & {
+    method: string;
+    args?: any[];
+    parseResultXdr: (xdr: string | xdr.ScVal | Err) => T;
+  };
+
+export const NULL_ACCOUNT = "GAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAWHF"
+
+export class AssembledTransaction<T> {
+  public raw: Tx
+  private simulation?: SorobanRpc.SimulateTransactionResponse
+  private simulationResult?: SorobanRpc.SimulateHostFunctionResult
+  private simulationTransactionData?: xdr.SorobanTransactionData
+  private server: Server
+
+  toJSON() {
+    return JSON.stringify({
+      method: this.options.method,
+      tx: this.raw?.toXDR(),
+      simulationResult: {
+        auth: this.simulationData.result.auth.map(a => a.toXDR('base64')),
+        retval: this.simulationData.result.retval.toXDR('base64'),
+      },
+      simulationTransactionData: this.simulationData.transactionData.toXDR('base64'),
+    })
+  }
+
+  static fromJSON<T>(
+    options: Omit<AssembledTransactionOptions<T>, 'args'>,
+    { tx, simulationResult, simulationTransactionData }:
+      {
+        tx: XDR_BASE64,
+        simulationResult: {
+          auth: XDR_BASE64[],
+          retval: XDR_BASE64,
+        },
+        simulationTransactionData: XDR_BASE64,
+      }
+  ): AssembledTransaction<T> {
+    const txn = new AssembledTransaction(options)
+    txn.raw = TransactionBuilder.fromXDR(tx, options.networkPassphrase) as Tx
+    txn.simulationResult = {
+      auth: simulationResult.auth.map(a => xdr.SorobanAuthorizationEntry.fromXDR(a, 'base64')),
+      retval: xdr.ScVal.fromXDR(simulationResult.retval, 'base64'),
+    }
+    txn.simulationTransactionData = xdr.SorobanTransactionData.fromXDR(simulationTransactionData, 'base64')
+    return txn
+  }
+
+  private constructor(public options: AssembledTransactionOptions<T>) {
+    this.server = new Server(this.options.rpcUrl, {
+      allowHttp: this.options.rpcUrl.startsWith("http://"),
+    });
+  }
+
+  static async fromSimulation<T>(options: AssembledTransactionOptions<T>): Promise<AssembledTransaction<T>> {
+    const tx = new AssembledTransaction(options)
+    const contract = new Contract(options.contractId);
+
+    tx.raw = new TransactionBuilder(await tx.getAccount(), {
+      fee: options.fee?.toString(10) ?? '100',
```

Comment:
using 

```suggestion
      fee: options.fee?.toString(10) ?? BASE_FEE,
```

from the SDK would be better than the magic number